test/integration/kubernetes_deploy_test.rb

```diff
@@ -616,6 +616,39 @@ def test_output_when_switching_labels_incorrectly
     ], in_order: true)
   end
 
+  def test_scale_down_deployment
```

Comment:
nit: `test_scale_existing_deployment_down_to_zero` (even though this isn't a regression test in the end, I'd still assume the "to zero" part is what made this scenario sensitive to the premature timeouts bug)

Btw how long does this test take to run?

Reply:
Updated the test function name.

Total time for the test is 7 sec in which the first deployment of 1 pod took 4.9sec and the scaledown to 0 pod took the rest 2.1 sec.